src/components/latency_test.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react"
+
+const states = {
+  INIT: "init",
+  TESTING: "testing",
+  TESTED: "tested",
+  ERROR: "error",
+}
+
+const getLatencyToCloudflare = async () => {
+  const TEST_URL = "https://1.1.1.1/cdn-cgi/trace"
+  const NUM_TESTS = 5
+
+  let started = 0
+  async function load() {
+    started++
+
+    await fetch(TEST_URL, {
+      mode: "no-cors",
+      cache: "no-store",
+    })
+
+    if (started <= NUM_TESTS) await load()
+  }
+
+  await load()
+  let entries = performance.getEntriesByName(TEST_URL, "resource")
+  let min = null
+
+  for (
+    let i = Math.max(0, entries.length - NUM_TESTS);
+    i < entries.length;
+    i++
+  ) {
+    // iOS Safari doesn’t report duration
+    const duration =
+      entries[i].duration || entries[i].responseEnd - entries[i].fetchStart
+
+    if (!min || duration < min) min = duration
+  }
+
+  return min | 0
+}
+
+export default () => {
+  const [state, setState] = useState(states.INIT)
+  const [latency, setLatency] = useState(null)
+
+  const testLatency = async () => {
+    try {
+      setState(states.TESTING)
+      const newLatency = await getLatencyToCloudflare()
+      if (isNaN(newLatency) || newLatency === 0 || newLatency > 50)
+        setLatency(null)
```

Comment:
I'm not super sure what this code does, it was ported from the original implementation as well. cc @adamschwartz as there may be an opportunity to refactor or better express this idea inside of this new implementation. Not a blocker at the moment